protocol/src/test/scala/io/mediachain/protocol/JournalBlockGenerators.scala

```diff
@@ -0,0 +1,193 @@
+package io.mediachain.protocol
+
+object JournalBlockGenerators {
+  import io.mediachain.protocol.Datastore._
+  import org.scalacheck._
+  import DataObjectGenerators._
+
+
+  case class JournalBlockWithDatastore(block: JournalBlock, datastore: InMemoryDatastore)
+  case class BlockChainWithDatastore(blockChain: List[JournalBlock], datastore: InMemoryDatastore)
+
+  /**
+    * Make a generator that returns a chain cell for the given `CanonicalRecord`
+    *
+    * @param canonical either an `Entity` or `Artefact` to generate a cell for
+    * @return depending on the type of `CanonicalRecord` given, a generator for
+    *         either a subtype of `EntityChainCell` or `EntityArtefactCell`.
+    *         Note that the `chain` field of the generated cell will be
+    *         set to `None`
+    *
+    */
+  def genCellFor(canonical: CanonicalRecord)
+  : Gen[ChainCell] =
+    canonical match {
+      case e: Entity => genEntityUpdateCell(e, genNilReference)
```

Comment:
it is?  I thought because `CanonicalRecord` was sealed you only had to match against the two concrete types.